R/core_gateway.R

```diff
@@ -0,0 +1,124 @@
+wait_connect_gateway <- function(gatewayAddress, gatewayPort, config, isStarting) {
+  waitSeconds <- if (isStarting)
+    spark_config_value(config, "sparklyr.gateway.start.timeout", 60)
+  else
+    spark_config_value(config, "sparklyr.gateway.connect.timeout", 1)
+
+  gateway <- NULL
+  commandStart <- Sys.time()
+
+  while (is.null(gateway) && Sys.time() < commandStart + waitSeconds) {
```

Comment:
Does this loop need a `Sys.sleep()` call? (do we want to wait for some interval after a failed attempt to create a `socketConnection()`, or is it okay to eagerly keep trying?)